### DISCUSSION TITLE
How to pre-load a script and allow script auto-upgrade ?

### DISCUSSION THREAD

**Discussion by ygelt** [arendst/Tasmota#20093] [Q&A]
Hi all, I use Tasmota scripting on ESP32 (for 8266 compatibility) and would like to : 
 1- Load the initial script inside the Tasmota32.bin file and start it on boot.
 2- Auto-update the script file from a http file and reboot.
Thanks by advance for any idea or solution :-)

**Comment by sfromis** [comment]
1) See https://github.com/arendst/Tasmota/pull/13153

**Reply by ygelt** [reply]
Many thanks, it works perfectly to pre-load and start the script after backslashing ", ? and \r\n, I used it with TasmoCompiler.

But how to auto-update the script ? I suppose the root file script.txt must be overwritten with a fetched one but this will probably lead to a script memory error ... is there a kind of "reboot on script fail" available ?
